log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -24,25 +26,51 @@
 class BufferedHandler
         extends Handler
 {
-    private static final int MAX_BATCH_COUNT = 1024;
+    private static final int MAX_BATCH_COUNT = 10240;
     private static final byte[] POISON_MESSAGE = new byte[0];
 
     private final MessageOutput messageOutput;
     private final Thread thread;
     private final BlockingQueue<byte[]> queue = new ArrayBlockingQueue<>(MAX_BATCH_COUNT);
     private final AtomicBoolean closed = new AtomicBoolean();
     private final AtomicLong droppedMessages = new AtomicLong(0);
+    private final PrintStream stdErr;
 
-    public BufferedHandler(MessageOutput messageOutput, Formatter formatter)
+    public BufferedHandler(MessageOutput messageOutput, Formatter formatter, PrintStream stdErr)
     {
         this.messageOutput = requireNonNull(messageOutput, "messageOutput is null");
+        this.stdErr = requireNonNull(stdErr, "stdErr is null");
         setFormatter(requireNonNull(formatter, "formatter is null"));
 
+        setErrorManager(new BufferedHandlerErrorManager());
+
         thread = new Thread(this::logging);
         thread.setName("log-writer");
         thread.setDaemon(true);
     }
 
+    private class BufferedHandlerErrorManager
+            extends ErrorManager
+    {
+        private boolean reported;
+
+        public synchronized void error(String msg, Exception ex, int code)
+        {
+            if (!reported) {
+                return;
+            }
+            reported = true;
+            String text = "java.util.logging.ErrorManager: " + code;
+            if (msg != null) {
+                text = text + ": " + msg;
+            }
+            stdErr.println(text);
+            if (ex != null) {
+                ex.printStackTrace(stdErr);
```

Comment:
Using `Throwables.getStackTraceAsString()` would be better, so that stderr is written to only once, potentially behaving better if there are other concurrent writes to stderr.